pandas/core/array_algos/take.py

```diff
@@ -90,6 +95,13 @@ def take_nd(
     if not isinstance(arr, np.ndarray):
         # i.e. ExtensionArray,
         # includes for EA to catch DatetimeArray, TimedeltaArray
+        if not is_1d_only_ea_obj(arr):
```

Comment:
can you change these conditions to be a positive one, eg. isinstance(..., (DatetimeArray, TimedeltaArray))?

Reply:
i can flip is_1d_only_ea_obj to is_2d_compat_ea_obj i guess; i like having a function for this so that a) it can potentially be expanded and b) it is easy to find where it is used

Reply:
i looked at this and it ended up making things more verbose